pkg/rbd/controllerserver.go

```diff
@@ -156,17 +158,24 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	volumeID := req.GetVolumeId()
 	rbdVol := &rbdVolume{}
 	if err := loadVolInfo(volumeID, path.Join(PluginFolder, "controller"), rbdVol); err != nil {
+		if os.IsNotExist(errors.Cause(err)) {
```

Comment:
```go
if os.IsNotExist(err)
```
see https://golang.org/pkg/os/#IsNotExist

Reply:
No, that doesn't work here. `err` is the result of `errors.Wrap` or (before this change) `fmt.Errorf`. For both `IsNotExist` returns false.

```
$ cat /tmp/test.go
package main

import (
	"fmt"
	"github.com/pkg/errors"
	"os"
)

func main() {
	_, err := os.Stat("/no/such/file")
	err2 := errors.Wrap(err, "stat")
	fmt.Printf("err: %v, err2: %v\n", os.IsNotExist(err), os.IsNotExist(err2))
}
$ go run /tmp/test.go
err: true, err2: false
```

Reply:
i see, `loadVolInfo` buries the root cause.